lib/rules/use-isnan.js

```diff
@@ -21,9 +21,17 @@ const astUtils = require("./utils/ast-utils");
  * @returns {boolean} `true` if the node is 'NaN' identifier.
  */
 function isNaNIdentifier(node) {
```

Comment:
This function is used in all checks, including those where the rule already provides suggestions, so the question now is should the rule provide suggestions when `NaN` is in a sequence expression, or should we also update the code that creates suggestions to skip those cases?

For example:

```js
/* eslint use-isnan: "error" */

x === (foo(), NaN);
```

After the suggestion is applied, the `foo()` call is lost:

```js
/* eslint use-isnan: "error" */

Number.isNaN(x);

```

Reply:
Ooh, yeah yeah you're right, totally forgot about it.
Yes, I think we can skip those cases, which is what I've done [here](
https://github.com/eslint/eslint/pull/18063/files#diff-f85ca2d533ed63955015070ff4e1b8d0e61e8a3b9efb316ac556ec699858f0d6R178-R182) already.

I don't think there is a safe way to keep those side effects, WDYT?

Reply:
> I don't think there is a safe way to keep those side effects, WDYT?

I agree, those cases should be fixed manually.